File: 2259-Remove-Digit-From-Number-to-Maximize-Result.go
func removeDigit(number string, digit byte) string {

	var waitDelete = 0

	for i, v := range number {

		if int(v) == int(digit) {

			if i < len(number)-1 && int(number[i+1]) > int(number[i]) {
				waitDelete = i
				break
			}

			waitDelete = i
		}

	}

	return number[:waitDelete] + number[waitDelete+1:]
}

File: 2259-Remove-Digit-From-Number-to-Maximize-Result.js
/**
 * @param {string} number
 * @param {character} digit
 * @return {string}
 */
var removeDigit = function (number, digit) {
    let numberList = number.split('');
    let length = numberList.length;
    let waitDelete = 0;
    for (let i = 0; i < length; i++) {
        if (numberList[i] == digit) {
            if (i < length - 1 && Number(numberList[i + 1]) > Number(numberList[i])) {
                waitDelete = i;
                break;
            }
            waitDelete = i;
        }
    }
    return numberList.filter((item, index) => index != waitDelete).join('');
};
